crates/librqbit/src/api.rs

```diff
@@ -320,3 +326,28 @@ fn make_torrent_details(
         files,
     })
 }
+
+fn torrent_file_mime_type(
+    info: &TorrentMetaV1Info<ByteBufOwned>,
+    file_idx: usize,
+) -> Result<&'static str> {
+    let file_name = info
+        .iter_filenames_and_lengths()?
+        .enumerate()
```

Comment:
Rather then enumerate().find_map(if ...) wouldn't just `nth(file_idx)` work?

Reply:
Absolutely,  bad me - cannot remember all those cool iterators combinators - fixed in next commit.